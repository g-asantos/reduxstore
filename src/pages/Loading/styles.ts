import styled from 'styled-components';

export const Container = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column-reverse;
  align-items: center;
  justify-content: center;
`;

export const AnimationContainer = styled.div`
  width: 50vh;
  height: 50vh;
`;

export const AnimationText = styled.p`
  font-weight: 700;
  font-size: 20px;
`;
