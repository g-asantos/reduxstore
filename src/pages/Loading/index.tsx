import Lottie from 'lottie-web';
import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { resetCart } from '../../store/modules/cart/actions';

import { AnimationContainer, AnimationText, Container } from './styles';

const Loading: React.FC = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  useEffect(() => {
    Lottie.loadAnimation({
      container: document.querySelector('#animation') as Element,
      renderer: 'svg',
      loop: true,
      autoplay: true,

      // eslint-disable-next-line global-require
      animationData: require('../../assets/35679-click-to-buy-basket.json'),
    });
  }, []);

  useEffect(() => {
    setTimeout(() => {
      dispatch(resetCart());

      history.push('/confirm');
    }, 4000);
  }, [history, dispatch]);

  return (
    <Container>
      <AnimationText>Your Order is processing...</AnimationText>
      <AnimationContainer id="animation" />
    </Container>
  );
};

export default Loading;
