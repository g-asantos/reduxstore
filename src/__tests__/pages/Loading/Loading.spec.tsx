import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../../../pages/Loading';

const mockedHistoryPush = jest.fn();
const mockedDispatch = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useHistory: () => ({
      push: mockedHistoryPush,
    }),
  };
});

jest.mock('react-redux', () => {
  return {
    useDispatch: () => ({
      dispatch: mockedDispatch,
    }),
  };
});

jest.mock('lottie-web', () => {
  return {
    Lottie: () => ({
      loadAnimation: jest.fn(),
    }),
  };
});

describe('Loading page', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Loading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
