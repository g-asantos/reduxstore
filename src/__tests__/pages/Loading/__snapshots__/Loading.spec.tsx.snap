// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading page snapshot renders 1`] = `
<div
  className="sc-bdnylx VPFCP"
>
  <p
    className="sc-dlnjPT eThZXH"
  >
    Your Order is processing...
  </p>
  <div
    className="sc-gtssRu VarYa"
    id="animation"
  />
</div>
`;
